pkg/image/importer/importer.go

```diff
@@ -129,6 +132,11 @@ func (i *ImageStreamImporter) importImages(ctx gocontext.Context, retriever Repo
 		repoName := defaultRef.RepositoryName()
 		registryURL := defaultRef.RegistryURL()
 
+		if !importer.isAllowedRegistry(defaultRef.RegistryHostname()) {
+			isi.Status.Images[i].Status = invalidStatus("", field.Invalid(field.NewPath("from", "name"), from.Name, fmt.Sprintf("registry %q not allowed in master configuration", defaultRef.RegistryHostname())))
```

Comment:
This message needs to be really good - this is going to break people.  You're going to have to make sure the CLI and UI do a good job of presenting this error.

```
Images may not be imported from %q - only images from %s are allowed
```